libraries/freertos_plus/standard/tls/src/iot_tls.c

```diff
@@ -880,7 +881,7 @@ BaseType_t TLS_Connect( void * pvContext )
         mbedtls_ssl_conf_ca_chain( &pxCtx->xMbedSslConfig, &pxCtx->xMbedX509CA, NULL );
 
         /* Configure the SSL context for the device credentials. */
-        xResult = prvInitializeClientCredential( pxCtx );
+        xPKCSResult = prvInitializeClientCredential( pxCtx );
```

Comment:
```
/* Put the result of loading device key and certifiacte in a separate
 * variable (xPKCSResult instead of xResult). The reason is that we
 * want to attempt TLS handshake even if the device key and certificate
 * are not loaded to support server only authentication. If the server
 * requires mutual authentication, the handshake will fail. */
 ```